lib/std/os/linux/riscv32.zig

```diff
@@ -0,0 +1,195 @@
+    return asm volatile ("ecall"
+        : [ret] "={x10}" (-> usize),
+        : [number] "{x17}" (@intFromEnum(number)),
+          [arg1] "{x10}" (arg1),
+          [arg2] "{x11}" (arg2),
+          [arg3] "{x12}" (arg3),
+        : "memory"
+    );
+}
+
+pub fn syscall4(number: SYS, arg1: usize, arg2: usize, arg3: usize, arg4: usize) usize {
+    return asm volatile ("ecall"
+        : [ret] "={x10}" (-> usize),
+        : [number] "{x17}" (@intFromEnum(number)),
+          [arg1] "{x10}" (arg1),
+          [arg2] "{x11}" (arg2),
+          [arg3] "{x12}" (arg3),
+          [arg4] "{x13}" (arg4),
+        : "memory"
+    );
+}
+
+pub fn syscall5(number: SYS, arg1: usize, arg2: usize, arg3: usize, arg4: usize, arg5: usize) usize {
+    return asm volatile ("ecall"
+        : [ret] "={x10}" (-> usize),
+        : [number] "{x17}" (@intFromEnum(number)),
+          [arg1] "{x10}" (arg1),
+          [arg2] "{x11}" (arg2),
+          [arg3] "{x12}" (arg3),
+          [arg4] "{x13}" (arg4),
+          [arg5] "{x14}" (arg5),
+        : "memory"
+    );
+}
+
+pub fn syscall6(
+    number: SYS,
+    arg1: usize,
+    arg2: usize,
+    arg3: usize,
+    arg4: usize,
+    arg5: usize,
+    arg6: usize,
+) usize {
+    return asm volatile ("ecall"
+        : [ret] "={x10}" (-> usize),
+        : [number] "{x17}" (@intFromEnum(number)),
+          [arg1] "{x10}" (arg1),
+          [arg2] "{x11}" (arg2),
+          [arg3] "{x12}" (arg3),
+          [arg4] "{x13}" (arg4),
+          [arg5] "{x14}" (arg5),
+          [arg6] "{x15}" (arg6),
+        : "memory"
+    );
+}
+
+const CloneFn = *const fn (arg: usize) callconv(.C) u8;
+
+pub extern fn clone(func: CloneFn, stack: usize, flags: u32, arg: usize, ptid: *i32, tls: usize, ctid: *i32) usize;
+
+pub const restore = restore_rt;
+
+pub fn restore_rt() callconv(.Naked) noreturn {
+    asm volatile (
+        \\ ecall
+        :
+        : [number] "{x17}" (@intFromEnum(SYS.rt_sigreturn)),
+        : "memory"
+    );
+}
+
+pub const MMAP2_UNIT = 4096;
+
+pub const F = struct {
+    pub const DUPFD = 0;
+    pub const GETFD = 1;
+    pub const SETFD = 2;
+    pub const GETFL = 3;
+    pub const SETFL = 4;
+    pub const GETLK = 5;
+    pub const SETLK = 6;
+    pub const SETLKW = 7;
+    pub const SETOWN = 8;
+    pub const GETOWN = 9;
+    pub const SETSIG = 10;
+    pub const GETSIG = 11;
+
+    pub const RDLCK = 0;
+    pub const WRLCK = 1;
+    pub const UNLCK = 2;
+
+    pub const SETOWN_EX = 15;
+    pub const GETOWN_EX = 16;
+
+    pub const GETOWNER_UIDS = 17;
+};
+
+pub const LOCK = struct {
+    pub const SH = 1;
+    pub const EX = 2;
+    pub const UN = 8;
+    pub const NB = 4;
+};
+
+pub const blksize_t = i32;
+pub const nlink_t = u32;
+pub const time_t = i64;
+pub const mode_t = u32;
+pub const off_t = i64;
+pub const ino_t = u64;
+pub const dev_t = u64;
+pub const blkcnt_t = i64;
+
+pub const timeval = extern struct {
+    tv_sec: time_t,
```

Comment:
See #20679 which renames these fields to `sec` and `usec`.

Reply:
:+1: Will make sure to rebase once that's merged.